Update tests and refactoring

brianz opened the pull request:
Why
---

- Continue refactoring and getting test coverage up higher

This change addresses the need by
---------------------------------

- Refactoring lots of things

Next Steps
----------

- Write a persistence backend for request/response session storage
- Write a test and possibly refactor resuming long-form audio

brianz committed:
Update tests and refactoring

Why
---

- Continue refactoring and getting test coverage up higher

This change addresses the need by
---------------------------------

- Refactoring lots of things

Next Steps
----------

- Write a persistence backend for request/response session storage
- Write a test and possibly refactor resuming long-form audio

brianz committed:
Add circleci config

Why
---

- Run tests on push
- Report test coverage during tests

This change addresses the need by
---------------------------------

- Adding `circleci.yml`

Next Steps
----------

- Get CodeClimate integrated

samcdavid commented on tests/conftest.py:
```diff
@@ -6,9 +6,12 @@
-os.environ['LAZYSUSAN_AWS_REGION'] = 'us-dummy-1'
+os.environ['LAZYSUSAN_SESSION_STORAGE_BACKEND'] = 'memory'
+os.environ['LAZYSUSAN_SESSION_DYNAMODB_TABLE_NAME'] = 'test'
+os.environ['LAZYSUSAN_SESSION_AWS_REGION'] = 'us-dummy-1'
```
`s/'/"/g`

samcdavid commented on tests/request_fixtures.py:
```diff
@@ -52,130 +52,98 @@ def build_request(request_type, request_intent=None, user_id=USER_ID):
+    request["request"].update({
+        "offsetInMilliseconds": offset_in_ms,
+        "token": token,
+        #"type": "AudioPlayer.PlaybackFinished"
```
We should probably nuke this.

brianz committed:
Cleanup

Why
---

- Clean things up...styling

This change addresses the need by
---------------------------------

- single -> double quotes
- remove comment
- add `.editorconfig`

Next Steps
----------

- None

brianz committed:
Ignore all things in tests dir for setup

Why
---

- Need to ignore *everything* in `tests` during setup

This change addresses the need by
---------------------------------

- Added to `exclude` list in `setup.py`

Next Steps
----------

- None
